src/core/polyhedral/codegen_llvm.cc

```diff
@@ -351,6 +355,45 @@ class CodeGen_TC : public Halide::Internal::CodeGen_X86 {
           << "[LLVM-IR] After optimization:\n"
           << toString(module.get());
     }
+
+    // In a separate process call the LLC and dump the ASM
+    // Since this is for debugging purposes only atm, separate process is fine.
+    if (FLAGS_llvm_dump_asm) {
+      // Create a temporary file (do not use racy tmpname;
+      // do not use tmpnam that immediately destroys the file).
+      char fileCName[256] = "/tmp/XXXXXXXXXX";
+      auto fd = mkstemp(fileCName);
+      ScopeGuard sg([fileCName]() { unlink(fileCName); });
+      CHECK_GE(fd, 0) << "mkstemp(): " << fileCName << " returned fd: " << fd;
+      close(fd);
+
+      // Close and reopen as FILE*
+      auto tmp = fopen(fileCName, "wb+");
```

Comment:
If you're worried about races, you should use `fdopen` instead of closing and
reopening.